File: frontend/src/app/modules/transactions/resources/models/category.ts
import { TransactionType } from "../../../dashboard/resources/models/transaction";

export interface Category{
  id: number,
  title: string,
  icon?: Uint8Array | string,
  type: TransactionType,
}

export interface CustomCategory{
  id: number,
  title: string,
  icon?: string,
  type: TransactionType,
  teamId: number,
  color: string
}

File: Finance.Domain/Entities/Category.cs
using System.Drawing;
using Finance.Domain.Common;

namespace Finance.Domain.Entities;

public class Category : BaseEntity
{
    public string Title { get; set; } = null!;
    public string Icon { get; set; } = null!;
    public TransactionType Type { get; set; }
    public string Color { get; set; } = null!;
}